### PR TITLE
Add cancan permissions for Admin::ManagerInvitationsController

### LINKED ISSUE #2197: Invite button not clickable when Inviting new Manager
## Description
When I use the Invite Manager field in Aus production, the box opens, I type the email of the manager I want to invite, but when I click the Invite button nothing happens. It works for super-admin users.

## Expected Behavior
The invite button should become clickable after I enter an email, and after I click it I should see a success/error message.

## Steps to Reproduce

1. Log in as enterprise user (not super admin).
1. Go to the Users tab of a profile you own. 
2. Click the invite manager button and enter an email, click Invite.

## Animated Gif/Screenshot
https://www.useloom.com/share/28d04bfcb06641079936c2f41790cb4c

## Context
I can't add managers.

## Severity
S3

## Your Environment
PC, Chrome.

## Possible fix

The POST request to invite a manager causes a 403 forbidden response. We probably need to tweak authorisations. See `app/models/spree/ability_decorator.rb`. This never worked. The issue was already there in the original feature introduced in 82457d2b632923118ac4aed171272a628ff3ed83.

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #2197

Minor amendment to previous PR, the implementation was changed a bit during code review, but the cancan permissions hadn't been updated to reflect that, which meant the manager invite wasn't working for non-superadmins.

#### What should we test?

Manager invites should now work correctly for non-superadmins.

#### Release notes

Manager invitation bugfix for #2197.

### CODE REVIEW

**Comment by myriamboure** [comment]
Can someone else quickly review that please so that we can test and merge? :point_up: @mkllnk or @HugsDaniel maybe?

File: app/controllers/admin/manager_invitations_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module Admin
   class ManagerInvitationsController < Spree::Admin::BaseController
+    authorize_resource class: false
```
Comment (mkllnk) [inline app/controllers/admin/manager_invitations_controller.rb]: This is the first time I see a line like this. What does it mean? I found https://github.com/ryanb/cancan/wiki/authorizing-controller-actions#authorize_resource explaining that it adds a before filter like `authorize!(:create, @manager_invitation || false)`. Why did you add `class: false`?

And why do we need two authorisations? One in this filter and the other one in the create method.

**Comment by enricostano** [comment]
> the implementation was changed a bit during code review, but the cancan permissions hadn't been updated to reflect that, which meant the manager invite wasn't working for non-superadmins.

@Matt-Yorkley do we have tests covering that behavior? Could you please add them? And in any case to the controller, right? Thanks!

File: app/controllers/admin/manager_invitations_controller.rb (hunk 1 above)
Comment (Matt-Yorkley) [reply app/controllers/admin/manager_invitations_controller.rb]: `authorize_class: false` seems to be the recommended way to use cancan in controllers that don't have a corresponding model? It tells cancan to look for symbol keys in the permissions list instead of a named class, and I think it skips some of the default behaviour like trying to instantiate a corresponding object on create actions, amongst others.

The additional permissions check inside the create action makes sure the user has permission to make changes to the specific enterprise in question (after we've pulled up that enterprise record) and not just general permission to access the create manager invites functions.

Comment (Matt-Yorkley) [reply app/controllers/admin/manager_invitations_controller.rb]: https://github.com/ryanb/cancan/wiki/Non-RESTful-Controllers#alternative-authorize_resource

**Comment by Matt-Yorkley** [comment]
@enricostano There's specs in the original PR: https://github.com/openfoodfoundation/openfoodnetwork/pull/2138. 

This PR doesn't change the functionality, it just fixes the missing permissions for non-superadmins.

**Comment by Matt-Yorkley** [comment]
Or do you want an extra spec to check the permissions? Maybe that would be useful.

**Comment by enricostano** [comment]
> Or do you want an extra spec to check the permissions? Maybe that would be useful.

@Matt-Yorkley that would be great

**Comment by mkllnk** [comment]
Staged on https://staging1.openfood.com.au/.

**Comment by sstead** [comment]
Testing notes @mkllnk 
- Both superadmin and enterprise owners can now invite new users to be managers
- There's one bug - when the new users confirms their email, they're taken to set a pw, and they get an error - 'password reset toek expired' . this is already on production, so not sure if it's reason to not push this PR.

https://docs.google.com/document/d/1P5nop3zhJgy0NTlrb9-CSr5pYdDcazNucr8GBGqx1iE/edit?usp=sharing

**Comment by sstead** [comment]
I've created this new issue for the 'reset password token expired' problem https://github.com/openfoodfoundation/openfoodnetwork/issues/2254

**Comment by sauloperez** [comment]
> Release notes

> Manager invitation bugfix for #2197.

They should be useful for humans :trollface: Consider describing the nature of the fix, please.
